File: ios/AppleMusicAuthModule.swift
import ExpoModulesCore
import MusicKit

public class AppleMusicAuthModule: Module {
    public func definition() -> ModuleDefinition {
        Name("AppleMusicAuth")
        
        Function("getAuthorizationStatus") { () -> String in
            return AppleMusicAuth.getAuthorizationStatus()
        }
        
        AsyncFunction("requestAuthorization") { () async throws -> String in
            return try await AppleMusicAuth.requestAuthorization()
        }
        
        AsyncFunction("setDeveloperToken") { (token: String) async throws in
            try AppleMusicAuth.setDeveloperToken(token)
        }

        AsyncFunction("getDeveloperToken") { (options: [String: Any]) async throws -> String in
            return try await AppleMusicAuth.getDeveloperToken(options)
        }

        AsyncFunction("getUserToken") { (options: [String: Any]) async throws -> String in
            return try await AppleMusicAuth.getUserToken(options)
        }

        AsyncFunction("clearTokenCache") { () async throws in
            AppleMusicAuth.clearTokenCache()
        }
    }
} 

File: src/AppleMusicAuthModule.ts
import { requireNativeModule, NativeModule } from "expo-modules-core";

import type { AppleMusicAuthStatus } from "./AppleMusicAuth.types";

export declare class AppleMusicAuthModule extends NativeModule {
  /**
   * Gets the current Apple Music authorization status
   * @returns Promise<AppleMusicAuthStatus>
   */
  readonly getAuthorizationStatus: () => Promise<AppleMusicAuthStatus>;

  /**
   * Requests Apple Music authorization from the user
   * @throws {AppleMusicAuthError}
   */
  readonly requestAuthorization: () => Promise<AppleMusicAuthStatus>;

  // Token Management
  readonly setDeveloperToken: (token: string) => Promise<void>;
  readonly getDeveloperToken: () => Promise<string>;
  readonly getUserToken: () => Promise<string>;
}

// This call loads the native module object from the JSI
export default requireNativeModule("AppleMusicAuth") as AppleMusicAuthModule;
